schema/crdb/dbinit.sql

```diff
@@ -1108,13 +1108,26 @@ CREATE TABLE IF NOT EXISTS omicron.public.oximeter (
     port INT4 CHECK (port BETWEEN 0 AND 65535) NOT NULL
 );
 
+/*
+ * The kind of metric producer each record corresponds to.
+ */
+CREATE TYPE IF NOT EXISTS omicron.public.producer_kind AS ENUM (
+    -- A sled agent for an entry in the sled table.
+    'sled_agent',
+    -- A service in the omicron.public.service table
+    'service',
+    -- A Propolis VMM for an instance in the omicron.public.instance table
+    'instance'
+);
+
 /*
  * Information about registered metric producers.
  */
 CREATE TABLE IF NOT EXISTS omicron.public.metric_producer (
     id UUID PRIMARY KEY,
     time_created TIMESTAMPTZ NOT NULL,
     time_modified TIMESTAMPTZ NOT NULL,
+    kind omicron.public.producer_kind,
```

Comment:
Same as for the Rust code: shouldn't this be required?

Reply:
I think the same answer applies here. Yes, I want to make this required, but I'm not sure how we can do that in one commit because of the producers outside the repository. Or at least, without introducing a commit in which Propolis and Dendrite will never be able to produce metrics, which might be fine.